airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -251,7 +251,7 @@ def execute_job(self, query: str, url: str) -> Tuple[Optional[str], Optional[str
         for i in range(0, self.MAX_RETRY_NUMBER):
             job_id = self.create_stream_job(query=query, url=url)
             if not job_id:
-                return None, None
+                return None, job_status
```

Comment:
Is it the fix to make the REST fallback work?

Reply:
Sorry for that, I should have described it better. [Here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py#L363) is the magic `if` statement which retries the read operation using REST. It only executes if job status is "Failed" after trying to create it. I believe this is the case when the job could not be created successfully so namely this status should be returned (it is initially defined with this value a couple of lines above).